From: SJ Park <sj@kernel.org>
Subject: samples/damon/mtier: do not stop first context for damon_start() failure
Date: Mon, 6 Jul 2026 07:06:17 -0700

damon_start() was able to fail while keeping the first context running. 
mtier hence stops the first context in the case.  damon_start() is
refactored to avoid that.  The error handling is no longer necessary. 
Remove it.

Link: https://lore.kernel.org/20260706140628.87414-3-sj@kernel.org
Signed-off-by: SJ Park <sj@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 samples/damon/mtier.c |    2 --
 1 file changed, 2 deletions(-)

--- a/samples/damon/mtier.c~samples-damon-mtier-do-not-stop-first-context-for-damon_start-failure
+++ a/samples/damon/mtier.c
@@ -193,8 +193,6 @@ static int damon_sample_mtier_start(void
 	if (!err)
 		return 0;
 
-	if (damon_is_running(ctxs[0]))
-		damon_stop(ctxs, 1);
 	damon_destroy_ctx(ctxs[0]);
 	damon_destroy_ctx(ctxs[1]);
 	return err;
_
